src/Modal.js

```diff
@@ -94,6 +94,13 @@ class Modal extends React.Component {
     /**
      * A callback fired when the escape key, if specified in `keyboard`, is pressed.
      */
+    onEscapeKeyDown: PropTypes.func,
+
+    /**
+     * Support for this function will be deprecated. Please use `onEscapeKeyDown` instead
+     * A callback fired when the escape key, if specified in `keyboard`, is pressed.
+     * @deprecated
+     */
     onEscapeKeyUp: PropTypes.func,
```

Comment:
use https://github.com/react-bootstrap/prop-types-extra#deprecatedvalidator-reason maybe? that way people will get the warning; makes it easier to remove for next version

we're already pulling in `prop-types-extra`